File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stk = []
    for c in s:
        if c == ')':
            if stk and stk[-1] == '(':
                stk.pop()
            else:
                stk.append(c)
        elif c == '(':
            stk.append(c)
    return False if stk else True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <vector>
#include <iostream>

using namespace std;

bool solution(string s)
{
    bool answer = true;
    vector<char> stk;
    for (auto c:s){
        if (c == '('){
            stk.push_back(c);
        }
        else if (c == ')'){
            if (!stk.empty() && stk.back() == '('){
                stk.pop_back();
            }
            else{
                stk.push_back(c);
            }
        }
    }
    if (!stk.empty()){
        return false;
    }
    else{
        return true;
    }
    return answer;
}